alphagov/paas-cf
# [#139179971] Upgrade CF to v253

[#139179971 Upgrade CF to >= v252](https://www.pivotaltracker.com/story/show/139179971)

What?
-----

We want to upgrade CF to v253.

In this PR we do so. Changes to highlight:

 * Added mutual TLS for cloudcontroller, tps, etc.
 * Prepare manifest to use bosh links in the future.
 * Upgraded to the latest stemcell in the recommended series.

We had to patch the cf-acceptance-tests in https://github.com/cloudfoundry/cf-acceptance-tests/pull/188 and we added a hardcoded checkout in the task to use our patched version.

Dependencies
--------------------

 * PR for the docker image to update the cf-client https://github.com/alphagov/paas-docker-cloudfoundry-tools/pull/91 Once merged, remove the temporary commit.

How to test?
-----------

 - Code review.
 - Review notes from pivotal story.
 - Deployment test:
   1. deploy from master
   2. deploy from this branch
   3. Check all tests pass

Who?
----

Anyone but @keymon or @csaliceti.
